gneiss/regression/_model.py

```diff
@@ -0,0 +1,187 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+from skbio.stats.composition import ilr_inv
+from gneiss._model import Model
+
+
+class RegressionModel(Model):
+    def __init__(self, *args, **kwargs):
+        """
+        Summary object for storing regression results.
+
+        A `RegressionResults` object stores information about the
+        individual balances used in the regression, the coefficients,
+        residuals. This object can be used to perform predictions.
+        In addition, summary statistics such as the coefficient
+        of determination for the overall fit can be calculated.
+
+        Parameters
+        ----------
+        submodels : list of statsmodels objects
+            List of statsmodels result objects.
+        basis : pd.DataFrame
+            Orthonormal basis in the Aitchison simplex.
+            Row names correspond to the leafs of the tree
+            and the column names correspond to the internal nodes
+            in the tree. If this is not specified, then `project` cannot
+            be enabled in `coefficients` or `predict`.
+        tree : skbio.TreeNode
+            Bifurcating tree that defines `basis`.
+        balances : pd.DataFrame
+            A table of balances where samples are rows and
+            balances are columns.  These balances were calculated
+            using `tree`.
+        """
+        super().__init__(*args, **kwargs)
+
+    def coefficients(self, project=False):
+        """ Returns coefficients from fit.
+
+        Parameters
+        ----------
+        project : bool, optional
+            Specifies if coefficients should be projected back into
+            the Aitchison simplex [1]_.  If false, the coefficients will be
+            represented as balances  (default: False).
+
+        Returns
+        -------
+        pd.DataFrame
+            A table of values where columns are coefficients, and the index
+            is either balances or proportions, depending on the value of
+            `project`.
+
+        References
+        ----------
+        .. [1] Aitchison, J. "A concise guide to compositional data analysis,
+           CDA work." Girona 24 (2003): 73-81.
+        """
+        coef = pd.DataFrame()
+
+        for r in self.results:
+            c = r.params
+            c.name = r.model.endog_names
+            coef = coef.append(c)
```

Comment:
I think this use-case should be ok.  This function should not be a heavily used function - so we shouldn't worry about it for the time-being.  If it turns out to be too slow - we can optimize it later.

Right now, I'm getting about 30s for models with 15k features - so I'm not inclined to optimize this for the time being.